### DISCUSSION TITLE
How can i custom training loop, sampling data and loss function in SFTTrainer

### DISCUSSION THREAD

**Discussion by anhnh2002** [huggingface/trl#1433] [Q&A] [answered]
I want to change training loop, sample data, loss func. Same in this repo https://github.com/nju-websoft/SCKD/tree/main.
How can i do it?

**Comment by younesbelkada** [comment, marked as answer]
hi @nguyenhoanganh2002 
Thanks for the issue! I think the easiest is to subclass `STTrainer` to create your own trainer and overwrite the desired methods
